### PR TITLE
[ISSUE-440]Clean expired hdfs files and keep one replication.

### LINKED ISSUE #440: Clean expired hdfs files and keep one replication.

### PR DESCRIPTION
**Pull request by FMX**
# [BUG]/[FEATURE] title

### Related issues.
Closes #440 

### How was this patch tested?

/cc @related-reviewer

/assign @main-reviewer

### CODE REVIEW

File: server-worker/src/main/java/com/aliyun/emr/rss/service/deploy/worker/storage/FileWriter.java [hunk 1]
```diff
@@ -80,6 +82,9 @@ public final class FileWriter implements DeviceObserver {
   private final PartitionType partitionType;
 
   private Runnable destroyHook;
+  // only will be true if it is on hdfs and deleted because of replication is ready.
+  private boolean deleted = false;
+  private boolean hasReplication = false;
```
Comment (waitinfuture) [inline server-worker/src/main/java/com/aliyun/emr/rss/service/deploy/worker/storage/FileWriter.java:87]: delete

File: server-worker/src/main/java/com/aliyun/emr/rss/service/deploy/worker/storage/FileWriter.java [hunk 2]
```diff
@@ -80,6 +82,9 @@ public final class FileWriter implements DeviceObserver {
   private final PartitionType partitionType;
 
   private Runnable destroyHook;
+  // only will be true if it is on hdfs and deleted because of replication is ready.
+  private boolean deleted = false;
```
Comment (waitinfuture) [inline server-worker/src/main/java/com/aliyun/emr/rss/service/deploy/worker/storage/FileWriter.java:86]: delete

File: server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/storage/StorageManager.scala [hunk 3]
```diff
@@ -131,6 +132,36 @@ private[worker] final class StorageManager(conf: RssConf, workerSource: Abstract
     None
   }
 
+  lazy val heartBeatCount = new AtomicLong()
```
Comment (waitinfuture) [inline server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/storage/StorageManager.scala:135]: delete this lines

File: server-worker/src/main/java/com/aliyun/emr/rss/service/deploy/worker/storage/FileWriter.java (hunk 2 above)
Comment (FMX) [reply server-worker/src/main/java/com/aliyun/emr/rss/service/deploy/worker/storage/FileWriter.java:85]: it's necessary.

File: server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/storage/StorageManager.scala [hunk 4]
```diff
@@ -380,6 +383,16 @@ private[worker] final class StorageManager(conf: RssConf, workerSource: Abstract
         }
       }
     }
+
+    if (hdfsFs != null) {
+      val iter = hdfsFs.listFiles(new Path(hdfsDir, RssConf.workingDirName(conf)), false)
+      while (iter.hasNext) {
+        val fileStatus = iter.next()
+        if (fileStatus.getModificationTime < expireTime) {
+          StorageManager.hdfsFs.delete(fileStatus.getPath, false)
```
Comment (waitinfuture) [inline server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/storage/StorageManager.scala:392]: should be true here?

File: server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/storage/StorageManager.scala [hunk 5]
```diff
@@ -21,8 +21,8 @@ import java.io.{File, IOException}
 import java.nio.charset.StandardCharsets
 import java.nio.file.{Files, Paths}
 import java.util
-import java.util.concurrent.{ConcurrentHashMap, Executors, ThreadPoolExecutor, TimeUnit}
-import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.{ConcurrentHashMap, Executors, LinkedBlockingQueue, ThreadPoolExecutor, TimeUnit}
```
Comment (waitinfuture) [inline server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/storage/StorageManager.scala:24]: unused import

File: server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/storage/StorageManager.scala [hunk 6]
```diff
@@ -21,8 +21,8 @@ import java.io.{File, IOException}
...
+import java.util.concurrent.atomic.{AtomicInteger, AtomicLong}
```
Comment (waitinfuture) [inline server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/storage/StorageManager.scala:25]: unused import

File: server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/storage/StorageManager.scala [hunk 7]
```diff
@@ -237,11 +238,8 @@ private[worker] final class StorageManager(conf: RssConf, workerSource: Abstract
       throw new IOException("No available working dirs!")
     }
 
-    val partitionId = location.getId
-    val epoch = location.getEpoch
-    val mode = location.getMode
-    val fileName = s"$partitionId-$epoch-${mode.mode()}"
-
+    val fileName = location.getFileName
+    val hasReplication = location.getPeer != null
```
Comment (waitinfuture) [inline server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/storage/StorageManager.scala:242]: unused

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Pull request merged**